tests/test_sync_versions.py

```
@@ -0,0 +1,128 @@
+    sync_versions.main()  # returning without SystemExit is exit code 0
+    assert {p.name: p.read_text() for p in root.iterdir()} == before
+
+
+def test_check_mode_with_drift_exits_nonzero_names_files_writes_nothing(
+    tmp_path, monkeypatch
+):
+    root = _fake_repo(tmp_path, "2.0.0", "1.0.0", "2.0.0")
+    _point_script_at(monkeypatch, root)
+    monkeypatch.setattr(sys, "argv", ["sync_versions.py", "--check"])
+    before = {p.name: p.read_text() for p in root.iterdir()}
+    with pytest.raises(SystemExit, match="version drift.*plugin.json") as excinfo:
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)